src/lib/checkbox/checkbox.spec.ts

```diff
@@ -141,6 +141,38 @@ describe('MdCheckbox', () => {
 
     }));
 
+    it('should not set indeterminate to false when checked is set programmatically', async(() => {
+      testComponent.isIndeterminate = true;
+      fixture.detectChanges();
+
+      expect(checkboxInstance.indeterminate).toBe(true);
+      expect(inputElement.indeterminate).toBe(true);
+      expect(testComponent.isIndeterminate).toBe(true);
+
+      testComponent.isChecked = true;
+      fixture.detectChanges();
+
+      fixture.whenStable().then(() => {
+        fixture.detectChanges();
```

Comment:
Are the `detectChanges` calls necessary after `whenStable`? Also you may be able to avoid the nesting and the `whenStable` calls by wrapping the test with `fakeAsync`, instead of `async`.